### ISSUE TITLE
[bug] ["@eslint-react/naming-convention/use-state"] works in jsx files only

### ISSUE THREAD

**Issue by danielrentz** [Rel1cx/eslint-react#953]
### Describe the bug

Rule "@eslint-react/naming-convention/use-state" triggers in JSX/TSX files, but not in JS/TS files (hook implementations).

### Reproduction

https://stackblitz.com/edit/vitejs-vite-cu2wkgxw

- run `npm run lint`
- triggers in `App.tsx`
- should trigger in `useTest.ts` too

### Expected behavior

Should trigger in `useTest.ts` too.

### Platform and versions

```Text
"@eslint-react/eslint-plugin": "^1.28.0"
```

### Stack trace

```Text

```
